### PR TITLE
tasks: add retry task

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/4753.

This is to enable retry pulling `go` docker image.

### CODE REVIEW

File: pkg/tasks/spec.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package tasks
```
Comment (indyfree) [inline pkg/tasks/spec.go]: Nice to move it here :)

File: pkg/tasks/retry_task.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	t := func(ctx context.Context) error {
+		o := func() error {
+			err := config.Underlying(ctx)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			return nil
+		}
+		b := backoff.NewExponential(backoff.ShortMaxWait, backoff.ShortMaxInterval)
```
Comment (indyfree) [inline pkg/tasks/retry_task.go]: Any reason why you choose `backoff.ShortMaxWait, backoff.ShortMaxInterval` in particular? Does it make sense to make it configurable at the Task level?

Comment (kopiczko) [reply pkg/tasks/retry_task.go]: No reason. That made sense for my particular use-case. If we need to configure that we can add that to config and default to the short.

**Review by indyfree — APPROVED** [review]
LGTM

**Review by tuommaki — APPROVED** [review]
LGTM

**Pull request merged**